lit_nlp/client/services/deltas_service.ts

```diff
@@ -0,0 +1,151 @@
+    return outputSpecKeys.flatMap(specKey => {
+      const fieldNames = findSpecKeys(modelSpec.output, [specKey]);
+      return fieldNames.map(fieldName => ({modelName, specKey, fieldName}));
+     });
+  }
+
+  // Get a list of each time a generator was run, and the data points generated
+  public deltaInfoFromSource(source: Source): DeltaInfo {
+    const byGeneration: {[generationKey: string]: IndexedInput[]} = {};
+    this.appState.generatedDataPoints.forEach((d: IndexedInput) => {
+      const key = d.meta.creationId;
+      byGeneration[key] = (byGeneration[key] || []).concat([d]);
+    });
+
+    const deltaRows = Object.keys(byGeneration).flatMap(generationKey => {
+      const ds = byGeneration[generationKey];
+      return this.deltaRowsForSource(source, ds)
+    });
+    return {
+      generationKeys: Object.keys(byGeneration),
+      deltaRows
+    };
+  } 
+
+  private deltaRowsForSource(source: Source, ds: IndexedInput[]): DeltaRow[] {
+    const scoreReaders = this.getScoreReaders(source);
+    return scoreReaders.flatMap(scoreReader => {
+      return this.readDeltaRows(ds, source, scoreReader);
+    });
+  }
+
+  private readDeltaRows(ds: IndexedInput[], source: Source, readScore: ScoreReader): DeltaRow[] {
+    return ds.flatMap(d => {
+      const parent = this.appState.getCurrentInputDataById(d.meta.parentId);
+      if (parent == null) return [];
+      
+      const before = readScore(parent.id);
+      const after = readScore(d.id);
+      const delta = (before != null && after != null)
+        ? after - before
+        : undefined;
+      const deltaRow: DeltaRow = {before,after,delta,d,parent};
+      return [deltaRow];
+    });
+  }
+
+  private getScoreReaders(source: Source): ScoreReader[] {
+    const {modelName, specKey, fieldName} = source;
+
+    // Check for regression scores
+    if (specKey === 'RegressionScore') {
+      const readScoreForRegression: ScoreReader = id => {
+        return this.regressionService.regressionInfo[id]?.[modelName]?.[fieldName]?.prediction;
+      };
+      return [readScoreForRegression];
+    }
+
+    // Also support multiclass for multiple classes or binary
+    if (specKey === 'MulticlassPreds') {
+      const spec = this.appState.getModelSpec(modelName);
+      const predictionLabels = spec.output[fieldName].vocab!;
+      const margins = this.classificationService.marginSettings[modelName] || {};
+
+      const nullIdx = spec.output[fieldName].null_idx;
+      if (predictionLabels.length === 2 && nullIdx != null) {
+         const readScoreForMultiClassBinary: ScoreReader = id => {
+           return this.classificationService.classificationInfo[id]?.[modelName]?.[fieldName]?.predictions[1 - nullIdx];
+        };
+        return [readScoreForMultiClassBinary];
+      }
+
+      // Multiple classes for multiple tables.
+      predictionLabels.map((predictionLabel, index) => {
+        const readScoreForMultipleClasses: ScoreReader = id => {
```

Comment:
so one table per class in multiclass?

Reply:
Yeah, but visually it's just a single table with arrows to navigate in the upper right (eg, the second image in the PR description).  We could alternately scope it down to be less general and only apply to binary multiclass or regression.

I think showing the change in prediction across all classes for a multiclass prediction problem may need a different visual.

Reply:
I updated the UX here a bit; instead of paging through arrows there's a dropdown to pick which `field:predictionLabel` to look at.

I didn't look at a different visual for multiclass; I think using the dropdown keeps the UX more general but will also take a look at how to address this in the next PR (eg, for MNLI).